source/extensions/common/async_files/async_file_manager_factory.cc

```diff
@@ -51,10 +55,13 @@ std::shared_ptr<AsyncFileManager> AsyncFileManagerFactoryImpl::getAsyncFileManag
       it = managers_
                .insert({config.id(), std::make_shared<AsyncFileManagerThreadPool>(config, posix)})
                .first;
+      configs_.insert({config.id(), config});
       break;
     default:
       PANIC("unrecognized AsyncFileManagerConfig::ManagerType");
     };
+  } else if (!Protobuf::util::MessageDifferencer::Equivalent(configs_[config.id()], config)) {
+    PANIC("AsyncFileManager mismatched config");
```

Comment:
This should fail gracefully during configuration load, so I would throw an EnvoyException here and make sure you have a test for it.

Reply:
Done (and for the other panics).